### PR TITLE
Render flags for dynamic command help

### PR DESCRIPTION
**Pull request by crufter**
Added flag rendering, changed code to keep casing so `getservice` became `getService` since `getservice` produced a `Endpoint Registry.Getservice not found for service registry` anyway.

Right now I'm having a look at adding comments but this can be merged.

```sh
$ micro registry --help
NAME:
	micro registry

VERSION:
	latest

USAGE:
	micro registry [command]

COMMANDS:
	deregister
	getService
	listServices
	register
	watch

```

```sh
$ micro registry getService --help
NAME:
	micro registry getService

USAGE:
	micro registry getService [flags]

FLAGS:
	--service string
	--options_ttl int64
	--options_domain string
```

### CODE REVIEW

**Comment by asim** [comment]
Add a doc as part of this PR that explains how the dynamic cli works. Create docs/reference/cli.md or similar.

**Comment by crufter** [comment]
@asim extended the getting started to show usage but a reference is a very good idea. Doing

File: docs/README.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ with the complexity of distributed systems and provides simpler programmable abs
 Documentation, guides and quick starts for Micro
 
 - [Getting Started](getting-started) - The helloworld quickstart guide
+- [Reference](reference) - External resources for reference
```
Comment (asim) [inline docs/README.md:23]: In-depth reference for Micro CLI and services

Comment (crufter) [reply docs/README.md:23]: Ah copypaste fail! Thanks

**Review by asim — APPROVED** [review]

**Pull request merged**